integrations/oci/oci/README.md

```diff
@@ -0,0 +1,43 @@
+# OCI integration module
+
+This Helidon module requires Service Registry.
+
+The module uses (internally) a service of type `OciConfig`. This instance is used to configure OCI integration.
+Note that this service can be customized, if a provider with higher weight is created.
+The default service implementation uses environment variables, system properties, and a configuration file `oci-config.yaml` on file system, or on classpath (Weight: default - 10). 
+
+This module provides two services that can be used by other modules.
+
+## Authentication Details Provider
+
+Any service can have a dependency on `com.oracle.bmc.auth.AbstractAuthenticationDetailsProvider`, and it can be
+looked up using Service Registry lookup methods.
+
+The provider is looked up by using instances of `io.helidon.integrations.oci.spi.OciAtnStrategy`. The first service instance to provide an authentication provider is used, and no other service instance is called.
+
+The following out-of-the-box implementations exist:
+
+- Config based provider: uses `OciConfig` to create authentication provider; Weight: default - 10
+- Config file based provider: uses OCI config file (i.e. `~/.oci/config`) to create authentication provider; both file location and profile can be configured through `OciConfig`, Weight: default - 20
+- Resource principal provider: uses resource principal authentication provider; Weight: default - 30
+- Instance principal provider: uses instance principal authentication provider; Weight: default - 40
+
+To create a custom instance of authentication details provider, just create a new service for service registry
```

Comment:
We need to sync classes added in this module with what we already have in https://github.com/helidon-io/helidon/tree/main/integrations/oci/sdk/cdi/src/main/java/io/helidon/integrations/oci/sdk/cdi